### ISSUE TITLE
Network not set on invoice

### ISSUE THREAD

**Issue by cryptoquick** [RGB-WG/rgb-wallet#103]
Despite setting the network argument, we get still invoices like this:
`rgb:Cbw1h3zbHgRhA6sxb4FS3Z7GTpdj9MLb7Do88qh5TUH1/RGB20/1+utxob02FKN7XB4N5LWc1d6cXfkByZ8d1jB4BZhee1vdU7MwffTtLkiK3`
When parsed, it lacks the network argument.
This will be helpful for users to better verify and troubleshoot their transfers.

**Comment by dr-orlovsky** [comment]
The network is inside the contract id: contract genesis commits to the network. So on a wrong network it will be impossible to do a payment since the contract would not be found

**Comment by dr-orlovsky** [comment]
Makes sense. Any ideas for the best form for providing explicit network information in the current invoice format? Must it be made required (thus breaking backward compatibility of the past invoices)?

**Comment by cryptoquick** [comment]
Not required, most fields on the invoice struct are Option already, and so long as it can be deserialized in the struct, it doesn't need to be user-readable necessarily... Though, perhaps as a convention, testnet invoices could be prefixed with `trgb`.

**Comment by dr-orlovsky** [comment]
I think we can say that if the network is not provided it defaults to mainnet.

I am not sure about using custom URL scheme for testnet since it increases complexity for wallet registration with OS. I rather propose query parameter:

`rgb:contractId/RGB20?net=signet`

**Comment by cryptoquick** [comment]
That could work!

**Comment by nicbus** [comment]
Having an optional query parameter seems better because it retains backwards compatibility and defaulting to mainnet when not provided should be fine.

**Comment by dr-orlovsky** [comment]
Done my version in https://github.com/RGB-WG/rgb-wallet/pull/104

**Comment by dr-orlovsky** [comment]
The final solution had to combine both testnet info with the chain info (due to added support for Liquid and more future layer1).

https://github.com/RGB-WG/rgb-std/pull/118
